homeassistant/config_entries.py

```diff
@@ -527,6 +541,55 @@ def async_entries(self, domain: Optional[str] = None) -> List[ConfigEntry]:
                 options=entry.get('options'))
             for entry in config['entries']]
 
+    async def async_setup(self, entry_id: str) -> bool:
+        """Set up a config entry.
+
+        Return True if entry has been successfully loaded.
+        """
+        entry = self.async_get_entry(entry_id)
+
+        if entry is None:
+            raise UnknownEntry
+
+        if entry.state != ENTRY_STATE_NOT_LOADED:
+            raise OperationNotAllowed
+
+        # Setup Component if not set up yet
+        if entry.domain in self.hass.config.components:
+            await entry.async_setup(self.hass)
+        else:
```

Comment:
Aren't we missing a path to `entry.async_setup` in the else condition?  The component get's loaded but not the entry (yet we set it's state to loaded).

Perhaps it should be:
```python
        if entry.domain not in self.hass.config.components:
            result = await async_setup_component(
                self.hass, entry.domain, self._hass_config)

            if not result:
                return result

        await entry.async_setup(self.hass)
        return entry.state == ENTRY_STATE_LOADED
```

Reply:
Setting up a component will also set up all related config entries.

Reply:
I added a comment to clarify.